pkg/controllers/controlplanemachinesetgenerator/controller_test.go

```diff
@@ -1296,3 +1296,574 @@ var _ = Describe("controlplanemachinesetgenerator controller on Azure", func() {
 		})
 	})
 })
+
+var _ = Describe("controlplanemachinesetgenerator controller on GCP", func() {
+
+	var (
+		usEast1aFailureDomainBuilderGCP = resourcebuilder.GCPFailureDomain().WithZone("us-east-1a")
+
+		usEast1bFailureDomainBuilderGCP = resourcebuilder.GCPFailureDomain().WithZone("us-east-1b")
+
+		usEast1cFailureDomainBuilderGCP = resourcebuilder.GCPFailureDomain().WithZone("us-east-1c")
+
+		usEast1dFailureDomainBuilderGCP = resourcebuilder.GCPFailureDomain().WithZone("us-east-1d")
+
+		usEast1eFailureDomainBuilderGCP = resourcebuilder.GCPFailureDomain().WithZone("us-east-1e")
+
+		usEast1aProviderSpecBuilderGCP = resourcebuilder.GCPProviderSpec().WithZone("us-east-1a")
+
+		usEast1bProviderSpecBuilderGCP = resourcebuilder.GCPProviderSpec().WithZone("us-east-1b")
+
+		usEast1cProviderSpecBuilderGCP = resourcebuilder.GCPProviderSpec().WithZone("us-east-1c")
+
+		usEast1dProviderSpecBuilderGCP = resourcebuilder.GCPProviderSpec().WithZone("us-east-1d")
+
+		usEast1eProviderSpecBuilderGCP = resourcebuilder.GCPProviderSpec().WithZone("us-east-1e")
+
+		cpms3FailureDomainsBuilderGCP = resourcebuilder.GCPFailureDomains().WithFailureDomainBuilders(
+			usEast1aFailureDomainBuilderGCP,
+			usEast1bFailureDomainBuilderGCP,
+			usEast1cFailureDomainBuilderGCP,
+		)
+
+		cpms5FailureDomainsBuilderGCP = resourcebuilder.GCPFailureDomains().WithFailureDomainBuilders(
+			usEast1aFailureDomainBuilderGCP,
+			usEast1bFailureDomainBuilderGCP,
+			usEast1cFailureDomainBuilderGCP,
+			usEast1dFailureDomainBuilderGCP,
+			usEast1eFailureDomainBuilderGCP,
+		)
+
+		cpmsInactive3FDsBuilderGCP = resourcebuilder.ControlPlaneMachineSet().
+						WithState(machinev1.ControlPlaneMachineSetStateInactive).
+						WithMachineTemplateBuilder(
+				resourcebuilder.OpenShiftMachineV1Beta1Template().
+					WithProviderSpecBuilder(
+						usEast1aProviderSpecBuilderGCP.WithZone("").WithMachineType("n1-standard-4"),
```

Comment:
If you look at the other tests (aws/azure), this type is different
```suggestion
						usEast1aProviderSpecBuilderGCP.WithZone("").WithMachineType("n1-standard-8"),
```

Reply:
Well, once this gets changed, the pass that didn't pass passes. Do you understand that? If yes, would you mind explaining?

Reply:
Yes, the reason it wasn't passing is because, the test case expects that the CPMS it was creating had a different instance type to the machines that were created for the test. When the CPMS sees this, it should recreate the CPMS and therefore, change the object (delete and recreate it)

Because you had the same instance type everywhere, it wasn't detecting the outdated spec as it was meant to be

Reply:
Ok, I get it now. Thanks:-)